compiler/ballerina-lang/src/main/java/io/ballerina/projects/Module.java

```diff
@@ -181,6 +210,46 @@ Modifier updateDocument(DocumentContext newDocContext) {
             return this;
         }
 
+        /**
+         * Creates a copy of the existing module and adds a new resource to the new module.
+         *
+         * @param resourceConfig configurations to create the resource
+         * @return an instance of the Module.Modifier
+         */
+        public Modifier addResource(ResourceConfig resourceConfig) {
+            ResourceContext newResourceContext = ResourceContext.from(resourceConfig);
+            this.resourceContextMap.put(newResourceContext.documentId(), newResourceContext);
+            return this;
+        }
+
+        /**
+         * Creates a copy of the existing module and adds a new test resource to the new module.
+         *
+         * @param resourceConfig configurations to create the test resource
+         * @return an instance of the Module.Modifier
+         */
+        public Modifier addTestResource(ResourceConfig resourceConfig) {
+            ResourceContext newResourceContext = ResourceContext.from(resourceConfig);
+            this.testResourceContextMap.put(newResourceContext.documentId(), newResourceContext);
+            return this;
+        }
+
+        /**
+         * Creates a copy of the existing module and removes the specified resource from the new module.
+         *
+         * @param documentId documentId of the resource to remove
+         * @return an instance of the Module.Modifier
+         */
+        public Modifier removeResource(DocumentId documentId) {
```

Comment:
Is it better to split this method into two? `removeResource` and `removeTestResource`

Reply:
For bal documents, we have only `removeDocument` for both source files and test files. So followed the same approach for resources. Should we modify this only or removeDocument for the sake of consistency?

Reply:
Let's keep `removeDocument` then.